velox/type/Variant.cpp

```diff
@@ -221,11 +221,12 @@ std::string variant::toJson(const TypePtr& type) const {
       std::string b{};
       b += "[";
       bool first = true;
+      uint32_t idx = 0;
       for (auto& v : row) {
         if (!first) {
           b += ",";
         }
-        b += v.toJson();
+        b += v.toJson(type->childAt(idx++));
```

Comment:
We need to verify that the typeKind of `type->childAt(idx++)` matches typeKind of `v`

Reply:
Added the check. `VELOX_USER_CHECK_EQ(
            v.kind(), type->childAt(idx)->kind(), "Not same TypeKind");`